webapp/utils/aws/ecs.py

```diff
@@ -0,0 +1,121 @@
+import logging
+import re
+import time
+from typing import TYPE_CHECKING, Literal
+
+import boto3
+from attrs import define, field
+
+if TYPE_CHECKING:
+    from mypy_boto3_ecs.client import ECSClient as ECSClientType
+
+from webapp import Config
+from webapp.exceptions import ECSTaskDefinitionDoesNotExistError
+
+logger = logging.getLogger(__name__)
+
+
+@define
+class ECSClient:
+    """ECS Client for running and monitoring task runs."""
+
+    cluster: str = field(factory=lambda: Config().ALMA_SAP_INVOICES_ECS_CLUSTER)
```

Comment:
Just curious since I haven't used it before, what does the `field(factory=lambda:` pattern do here? Is this related to the `@property` decorator?

Reply:
Spent some time reading [documentation on `attrs.Factory` ](https://www.attrs.org/en/stable/api.html#attrs.Factory) but I think @ghukill described it best above in regards to "why `field(default=Config().<ENV>` doesn't work for our unit tests":

> whenever the class (`webapp.config.Config` is first imported / whatever the env vars are at that moment will be the class defaults for those properties forever after, regardless of monkeypatching)

I prefer the current approach over using `__attrs_post_init__` (as shown in @ghukill 's example above); the latter, as the doc describes, 

> **runs** after attrs is done initializing your instance. This is useful if you want to **derive some attribute from others** or perform some kind of **validation over the whole instance**.

I interpret retrieval of the environment 
variables as _part of_ / _the_ initialization step. Conceptually, the use of `__attrs_post_init__` might be confusing. 🤔 

Let me know your final thoughts, @ghukill and @ehanson8 !

Reply:
@jonavellecuerdo I agree that it is a part of initialization